tests/app/questionnaire/test_path_finder.py

```diff
@@ -535,3 +536,253 @@ def test_new_remove_answer_and_block_if_routing_backwards(list_store):
+        ),
+        (
+            "Yes",
+            "Yes",
+            "test_routing_and_skipping_section_dependencies",
+            "primary-person",
+            ["name-block"],
+        ),
+        (
+            "Yes",
+            "No",
+            "test_routing_and_skipping_section_dependencies",
+            "primary-person",
+            ["name-block", "age"],
+        ),
+        (
+            "No",
+            "Yes",
+            "test_routing_and_skipping_section_dependencies",
+            "primary-person",
+            ["name-block", "age", "reason-no-confirmation"],
+        ),
+    ),
+)
+def test_routing_path_block_ids_dependent_on_when_rules(
+    list_store,
+    skip_age_answer,
+    skip_confirmation_answer,
+    schema,
+    section_id,
+    expected_route,
+):
+    # Given
+    schema = load_schema_from_name(schema)
+    answer_store = AnswerStore()
+    answer_store.add_or_update(
+        Answer(answer_id="skip-age-answer", value=skip_age_answer)
+    )
+
+    progress = [
+        {
+            "section_id": "skip-section",
+            "list_item_id": None,
+            "status": CompletionStatus.COMPLETED,
+            "block_ids": ["skip-age"],
+        }
+    ]
+
+    if skip_confirmation_answer:
+        answer_store.add_or_update(
+            Answer(answer_id="skip-confirmation-answer", value=skip_confirmation_answer)
+        )
+        progress.append(
+            {
+                "section_id": "skip-confirmation-section",
+                "list_item_id": None,
+                "status": CompletionStatus.COMPLETED,
+                "block_ids": ["security", "skip-confirmation"],
+            }
+        )
+
+    progress_store = ProgressStore(progress)
+
+    # When
+    path_finder = PathFinder(
+        schema,
+        answer_store,
+        list_store,
+        progress_store,
+        {},
+        {},
+    )
+    routing_path = path_finder.routing_path(section_id=section_id)
+
+    # Then
+    expected_routing_path = RoutingPath(
+        expected_route,
+        section_id=section_id,
+    )
+    assert routing_path == expected_routing_path
+
+
+@pytest.mark.parametrize(
+    "skip_age_answer, schema, expected_route",
+    (
+        (
+            "Yes",
+            "test_new_routing_and_skipping_section_dependencies",
+            ["repeating-sex"],
+        ),
+        (
+            "No",
+            "test_new_routing_and_skipping_section_dependencies",
+            ["repeating-sex", "repeating-age"],
+        ),
+        (
+            "Yes",
+            "test_routing_and_skipping_section_dependencies",
+            ["repeating-sex"],
+        ),
+        (
+            "No",
+            "test_routing_and_skipping_section_dependencies",
+            ["repeating-sex", "repeating-age"],
+        ),
+    ),
+)
+def test_routing_path_block_ids_dependent_on_when_rules_repeating(
+    skip_age_answer,
+    schema,
+    expected_route,
+):
```

Comment:
Same deduplicate can be applied here:
```suggestion
@pytest.mark.parametrize(
    "skip_age_answer, expected_route",
    (
        (
            "Yes",
            ["repeating-sex"],
        ),
        (
            "No",
            ["repeating-sex", "repeating-age"],
        ),
    ),
)
@pytest.mark.parametrize(
    "schema",
    (
        [
            "test_new_routing_and_skipping_section_dependencies",
            "test_routing_and_skipping_section_dependencies",
        ]
    ),
)
def test_routing_path_block_ids_dependent_on_when_rules_repeating(
    schema,
    skip_age_answer,
    expected_route,
):
```